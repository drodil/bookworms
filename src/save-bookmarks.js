import { mkdirSync, writeFileSync } from 'fs';

const writeBookmarks = (directory, bookmarks) => {
    bookmarks.forEach(({filename, body})=> {
        try {
            mkdirSync(directory, { recursive: true});
            writeFileSync(`${trimTrailingSlash(directory)}/${filename}`, body)
        } catch (error) {
            // todo write test for this error condition
            console.error(error.message)
        }
    })
}

const trimTrailingSlash = (directory) => {
    return directory.replace(/\/$/, "");
}

export {writeBookmarks, trimTrailingSlash}
