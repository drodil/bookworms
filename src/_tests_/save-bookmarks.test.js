import { rmSync, readFileSync } from 'fs';
import {writeBookmarks, trimTrailingSlash} from '../save-bookmarks';

describe('saving bookmarks', () => {
    describe('writeBookmarks', () => {
        test('should write bookmark files', () => {
            const path = './tmp'
            const bookmarks = [{
                filename: 'browser.html',
                body: '<div>hello world</div>'
            },
            {
                filename: 'readme.md',
                body: '_hello world_'
            }];
            writeBookmarks(path, bookmarks);
            bookmarks.forEach(({filename})=> {
                const bookmark = readFileSync(`${path}/${filename}`, 'utf8')
                expect(bookmark).toMatchSnapshot();
            });
            rmSync(path, { recursive: true });
        })
    })
    describe('trimTrailingSlash', () => {
        test('should not change path without trailing slash', () => {
            expect(trimTrailingSlash('./demo/config')).toEqual('./demo/config')
        })
        test('should change path removing trailing slash', () => {
            expect(trimTrailingSlash('./demo/config/')).toEqual('./demo/config')
        })
    })
})