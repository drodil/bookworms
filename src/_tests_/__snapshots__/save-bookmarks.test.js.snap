// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`saving bookmarks writeBookmarks should write bookmark files 1`] = `"<div>hello world</div>"`;

exports[`saving bookmarks writeBookmarks should write bookmark files 2`] = `"_hello world_"`;
